lib/web_ui/lib/src/engine/text/font_collection.dart

```diff
@@ -96,40 +96,78 @@ class FontCollection {
 }
 
 /// Manages a collection of fonts and ensures they are loaded.
-class _FontManager {
+class FontManager {
   final List<Future<void>> _fontLoadingFutures = <Future<void>>[];
 
-  factory _FontManager() {
+  factory FontManager() {
     if (supportsFontLoadingApi) {
-      return _FontManager._();
+      return FontManager._();
     } else {
       return _PolyfillFontManager();
     }
   }
 
-  _FontManager._();
-
+  FontManager._();
+
+  /// Registers assets to Flutter Web Engine.
+  ///
+  /// Browsers and browesers versions differ siginificantly on how a valid font
+  /// family name should be formatted. Notable issues are:
+  /// Safari12 and Firefox crash if you create a [html.FontFace] with a font
+  /// family that is not correct CSS syntax. Font family names accepted on these
+  /// browsers, when wrapped it in quotes.
+  /// Additionally, for Safari12 to work [html.FontFace] name should be loaded
+  /// correctly on the first try.
+  /// A font in Chrome chrashes if a [html.FontFace] is loaded only with quotes.
+  /// Unlike Safari12 if a valid version is loaded afterwards it will show
+  /// that fonts normally.
+  /// In Safari 13 the [html.FontFace] should be loaded with unquoted family
+  /// names.
+  /// In order to avoid all these browser compatibility issues this method;
+  /// detects the family names that might cause a conflict, loads it with
+  /// quotes and loads it again without quotes. For all the other family names
+  /// [html.FontFace] is loaded only once.
+  /// See: https://developer.mozilla.org/en-US/docs/Web/CSS/font-family#Valid_family_names
+  /// See: https://drafts.csswg.org/css-fonts-3/#font-family-prop
   void registerAsset(
     String family,
     String asset,
     Map<String, String> descriptors,
   ) {
-    // Safari and Firefox crash if you create a [html.FontFace] with a font
-    // family that is not correct CSS syntax. To ensure the font family is
-    // accepted on these browsers, wrap it in quotes.
-    // See: https://drafts.csswg.org/css-fonts-3/#font-family-prop
-    if (browserEngine == BrowserEngine.webkit || browserEngine == BrowserEngine.firefox) {
-      family = "'$family'";
+    final String familyNameInQuotes = "'$family'";
+    // Regular expression to detect punctuations. For example font family
+    // 'Ahem!' falls into this category.
+    final RegExp punctuations = RegExp(r"[.,:;!`\/#\$\%\^&~\*=\-_(){}]");
```

Comment:
Instead of listing all the possible punctuation symbols, is it possible to use something like this: `RegExp(r"[^a-z0-9 ]", caseSensitive: false)`? Which matches anything other than an alphabet or digit or whitespace.

Reply:
I think we should just try to match the whole family name with a regex for CSS identifiers: https://developer.mozilla.org/en-US/docs/Web/CSS/custom-ident

so something like

`RegExp(r"[a-zA-Z][a-zA-Z0-9\-_]*")`

Reply:
@hterkelsen 
I think using one regex for all the matches will create maintance/readibility issues. The correct regex for a valid family is very long and hard to understand in the first look.

If we use:
`static final RegExp cssIdentifiers = RegExp(r"[a-zA-Z][a-zA-Z0-9\-_]*");`

This does not cover valid cases such as:
- Gill Sans Extrabold
- ahem ahem

If we add the whitespace and use: 

`static final RegExp cssIdentifiers = RegExp(r"[a-zA-Z][a-zA-Z0-9\-_\s]*");
`
this also does not recognize invalid cases such as:
- Ahem 1989
- Goudy Bookletter 1911

We can also add the digit starting token case using \b\d yet this will make the regex even longer.

Please let me know if you have a suggestion for simpler regex? Otherwise use 2 regex one  @mdebbar 's other is the digit check:

`static final RegExp cssIdentifiers = RegExp(r"[^a-z0-9\s]", caseSensitive: false);
`
`static final RegExp startWithDigit = RegExp(r"\b\d");
`

Note ([the documentation](https://developer.mozilla.org/en-US/docs/Web/CSS/font-family)):
_"Font family names must either be given quoted as strings, or unquoted as a sequence of one or more identifiers. This means that punctuation characters and digits at the start of each token must be escaped in unquoted font family names."_

Reply:
You could do:

`static final RegExp cssIdentifiers = RegExp(r"[a-zA-Z][a-zA-Z0-9\-_]*(\s[a-zA-Z][a-zA-Z0-9\-_]*)+");`

Reply:
Please check the new version :)

I believe it is very readable. It has two parts but each of them are very small.

Thanks for the review!